### PR TITLE
Enable pubspec checks

### PR DESCRIPTION
**Pull request by stuartmorgan**
Turns on the new pubspec checks for this repository, and updates all the pubspecs to conform.

Updates the version of flutter_plugin_tools used in order to pick up the new check.

This does not bump any versions; the metadata updates are minor, so can simply go out if/when the packages are next released.

Part of https://github.com/flutter/flutter/issues/82913

## Pre-launch Checklist

- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated `CHANGELOG.md` to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -167,7 +167,7 @@ linter:
     - slash_for_doc_comments
     # - sort_child_properties_last # not yet tested
     - sort_constructors_first
-    - sort_pub_dependencies
+    # - sort_pub_dependencies
```
Comment (ditman) [inline analysis_options.yaml:170]:
```suggestion
```

This was added in line 215, so remove commented-out line?

File: packages/cross_file/pubspec.yaml [hunk 2]
```diff
@@ -1,8 +1,13 @@
 name: cross_file
 description: An abstraction to allow working with files across multiple platforms.
 repository: https://github.com/flutter/packages/tree/master/packages/cross_file
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+animations%22
```
Comment (ditman) [inline packages/cross_file/pubspec.yaml:4]:
```suggestion
issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+cross_file%22
```

File: packages/metrics_center/pubspec.yaml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 name: metrics_center
-version: 0.1.0
 description:
   Support multiple performance metrics sources/formats and destinations.
-homepage:
+repository:
   https://github.com/flutter/packages/tree/master/packages/metrics_center
```
Comment (ditman) [inline packages/metrics_center/pubspec.yaml:5]:
```suggestion
repository: https://github.com/flutter/packages/tree/master/packages/metrics_center
```

Move to same line

**Review by ditman — APPROVED** [review]
LGTM, I added a couple of small suggestions.

File: analysis_options.yaml (hunk 1 above)
Comment (stuartmorgan) [reply analysis_options.yaml:170]: This is what's in the Flutter version that I hope to eventually actually share, so I wanted it to match.

File: packages/cross_file/pubspec.yaml (hunk 2 above)
Comment (stuartmorgan) [reply packages/cross_file/pubspec.yaml]: Good catch, I thought I did a final sweep for copypasta mistakes.

**Pull request merged**